### DISCUSSION TITLE
connect/merge two tracks

### DISCUSSION THREAD

**Discussion by ChenSun-YP** [mido/mido#443] [Q&A] [answered]
Hi , I have 5 midi file that are 5 parts of one song that I want to connect and form into 1 type 0 midi file(the full song), I tried with the following code but it will just stack the 5 tracks over in one track because the all start from` time = 0`, is there a way to merge/connect these tracks in a sequence that 5 parts will play in order and not cover others?

```

merged_mid = MidiFile(type=1)
merged_mid.ticks_per_beat = pattern_list[0].ticks_per_beat
for item in pattern_list:
  merged_mid.tracks = merged_mid.tracks  + item.tracks
  merge_tracks(merged_mid.tracks)
merged_mid.print_tracks()

merged_mid.tracks = [merge_tracks(merged_mid.tracks)]
merged_mid.print_tracks(meta_only=True)
merged_mid.save('merged.mid')
```

**Comment by rdoursenaud** [comment, marked as answer]
You have to offset all the messages from your parts with the previous part’s duration before merging.
